### ISSUE TITLE
Sourcemapping fails unpredictably

### ISSUE THREAD

**Issue by cefn** [vercel/next.js#26457]
### What version of Next.js are you using?

Tried 10.0.0 and 11.0.0

### What version of Node.js are you using?

12.20.2

### What browser are you using?

Chrome

### What operating system are you using?

Chrome OS

### How are you deploying your application?

yarn run dev

### Describe the Bug

# Issue description

I encounter unreliable sourcemap debugging when using Next. 

See below for detail of the Monorepo case, but I have found there are problems even in a standalone case, as per the followup comment in this issue which provides probably a simpler test case.

# Monorepo case

Some files mounted from my local filesystem source folder show that they are loaded and bound, others are not. The application is successfully bundled and runs, but cannot be reliably debugged. The results seem worse since the Webpack 5 upgrade, and now almost no files successfully bind breakpoints.

Breakpoints in visibly loaded and bound files (with a green circle) work OK. Breakpoints in files without the green circle silently fail.

The below screenshot shows the running app which includes all the files shown in the window, but `util.ts` and `view.tsx` ARE bound, while `domain.ts` and `game.ts` are mysteriously unbound and breakpoints placed in those files (like the one in the screenshot) are ignored. 

![image](https://user-images.githubusercontent.com/159819/122828275-5a541f80-d2dd-11eb-8a0a-d5a990dbacbd.png)

# Typescript and webpack config

I have previously removed `tsconfig.json` and `next.config.js` and `next-env.d.ts` and ran `touch tsconfig.json` then `yarn run dev` to trigger Next's auto-generation of typescript config, but the following changes were needed to get it to run correctly in the monorepo. Perhaps these changes might give an idea why sourcemapping is unpredictable in this monorepo config...

A line needed to be added to the autogenerated tsconfig `"extends": "../../tsconfig.json"` in order to pull in the typescript [package resolutions from the monorepo](https://github.com/cefn/lauf/blob/simplify-async-example/tsconfig.json) else it wasn't resolving those locally. I added `noEmit:false` and `outDir:"dist"` to override the monorepo's top leve config so that files were actually generated for `next` to pick up, else it complained they were empty.

A next.config.js with a webpack override for transpiling monorepo typescript imports needed to be created at https://github.com/cefn/lauf/blob/simplify-async-example/apps/nextjs-snake/next.config.js else it was trying to parse the locally resolved typescript modules as javascript.

This seems enough to be able to compile and run the application, but since Next 11 none of the Filesystem source files can bind any breakpoints at all. 

### Expected Behavior

Source files which are correctly bundled and successfully run in the application should also have functioning sourcemaps when loaded through the Chrome inspector. 

### To Reproduce

* See followup comment for a simpler reproduction of sourcemapping problems using the same standalone project outside of a monorepo.
* See below for reproduction of the monorepo case.

The app ( https://github.com/cefn/lauf/tree/main/apps/nextjs-snake ) can be run by checking out the Lauf repository and running `yarn && cd apps/nextjs-snake && yarn run dev` then loading http://localhost:3000 and viewing the Sources tab of the Chrome inspector for the page. The file path where the source directory is checked out has to be added to the Chrome inspector workspace via the Sources > Filesystem tab to try and bind them and be able to insert breakpoints.

Both versions described below run and demonstrate the problem of unbindable breakpoints in source files.

# Version 10 and Webpack 4

The demonstration 'snake' example in the main branch (currently at [commit 2b05b762](https://github.com/cefn/lauf/tree/2b05b762198e627489f8128521619f7e9989d232/apps/nextjs-snake)) compiles and runs but only breakpoints in certain files using webpack4 and next 10.0.0. 

# Version 11 and Webpack 5

A branch at [898f03e7](https://github.com/cefn/lauf/tree/898f03e7136cbdc1991767197cd5d7e15bdaaa03/apps/nextjs-snake) upgrades React and NextJS to latest (17.0.2 and 11.0.0). It also compiles and runs but will never breakpoint.

**Comment by cefn** [comment]
For reference I created a standalone app separate from the monorepo at https://github.com/cefn/nextjs-snake-debug 

It has newer code based on async instead of generators. I am trying to debug this code as the snake doesn't move when you press keys as expected. There are no compiler errors and no console errors, meaning I have to try and step through the code to find the issue.

This isolated nextjs app can be run by checking it out, running `yarn` then `yarn run dev` and visiting `http://localhost:3000`

It seems to have all the source files bound and can accept some breakpoints in the Chrome inspector, but many attempted breakpoints are moved to the top of an async function declaration when you insert them, and source stepping usually doesn't function for more than one or two lines at a time.

For example I would like to step through the code from https://github.com/cefn/nextjs-snake-debug/blob/00229d5eb833f72e6838fd39996d816948323aff/src/game.ts#L46 but it is impossible to put a breakpoint there.

Other lines DO accept breakpoints but stepping one or two lines causes the stack 'pointer' to go to the top of an async method and stay there for tens of ticks, followed by leaping into transpiled code and (presumably) webpack libraries.

These behaviours make debugging of the async issue in the application impossible.

In this standalone example app I targeted ES2020 to minimise the amount of transpilation needed from Typescript, but ES5 and ES2015 also have the same problems.
